### PR TITLE
Handle errors correctly when logging fails

### PR DESCRIPTION
**Pull request by jonasfj**

### CODE REVIEW

**Review by isoos — APPROVED** [review]

File: lib/src/grpc_api_impl/logging_impl.dart [hunk 1]
```diff
@@ -371,17 +371,21 @@ class SharedLoggingService {
...
-      // stderr.  The logging daemon on the VM will make another attempt at
-      // uploading stderr via the logging API.
-      stderr.writeln('An error occured while writing log entries:\n'
-          'Error:$error\n'
-          '$stack');
-    }).whenComplete(() {
-      _outstandingRequests--;
-      _maybeClose();
-    });
+    unawaited(() async {
```
Comment (sigurdm) [inline lib/src/grpc_api_impl/logging_impl.dart:374]: I think this is the right way. We could also consider:
```
_clientStub.writeLogEntries(request).then((_) {}).catchError(....
```
As a way of making the types match.

**Review by sigurdm — APPROVED** [review]
LGTM

Could we check that the invalid_return_type_for_catch_error lint is enabled for our project?

File: lib/src/grpc_api_impl/logging_impl.dart (hunk 1 above)
Comment (jonasfj) [reply lib/src/grpc_api_impl/logging_impl.dart:374]: Yes, that could also have worked. but `try/catch` with `async/await` is just all around more sane :D

File: lib/src/grpc_api_impl/logging_impl.dart [hunk 2]
```diff
@@ -371,17 +371,21 @@ class SharedLoggingService {
...
+      try {
+        await _clientStub.writeLogEntries(request);
+      } catch (error, stack) {
+        // In case the logging API failed, we'll write the error message to
+        // stderr.  The logging daemon on the VM will make another attempt at
+        // uploading stderr via the logging API.
+        stderr.writeln('An error occured while writing log entries:\n'
```
Comment (sigurdm) [inline lib/src/grpc_api_impl/logging_impl.dart:381]: One more thing: should we emit the entry that we failed to log?

**Pull request merged**